pkg/manifests/configaware.go

```diff
@@ -208,20 +210,32 @@ func (ca *configAware) getResourcesByID(ctx context.Context) (map[string]resourc
 		if err != nil {
 			return nil, err
 		}
-		relConfigFilePath, err := cf.RelativeConfigPath()
+		resources, err := ca.manifests.ParseManifest(resourceManifests, cf.ConfigRelativeToWorkingDir())
 		if err != nil {
 			return nil, err
 		}
-		resources, err := ca.manifests.ParseManifest(resourceManifests, relConfigFilePath)
-		if err != nil {
-			return nil, err
-		}
-		for id, r := range resources {
-			if _, ok := resourcesByID[id]; ok {
-				return nil, fmt.Errorf("duplicate resource from %s and %s",
-					r.Source(), resourcesByID[id].resource.Source())
+
+		for id, generated := range resources {
+			if duplicate, ok := resourcesByID[id]; ok {
+				var duplicateErr error
+				switch {
+				case duplicate.configFile == cf:
+					// the duplicate is _also_ generated by this config file
+					duplicateErr = fmt.Errorf("duplicate definitions of resource %s generated by %s",
```

Comment:
Can you make the error messages consistent with https://github.com/fluxcd/flux/blob/6a0e353061f68ba902161af3b39b4e6557c23646/pkg/cluster/kubernetes/resource/load.go#L56 ?

Reply:
Yup